extensions/ql-vscode/src/view/results/results.tsx

```diff
@@ -164,76 +162,52 @@ export class ResultsApp extends React.Component<
 
   private updateStateWithNewResultsInfo(resultsInfo: ResultsInfo): void {
     this.setState((prevState) => {
-      const stateWithDisplayedResults = (
```

Comment:
Could we write tests for the behaviour before we made this change and then see if the same behaviour happens afterwards? That would mean we wouldn't need to guess if we've completely got it right.

Reply:
Is it hard because we would have to check visual components or because it's a timing issue?

Reply:
It's a timing issue, there's no real way to check the state in between the two `setState` calls, so we'd never get into the state that we'd want to test. We'd want to check what the view looks like in between [these two calls](https://github.com/github/vscode-codeql/blob/64d97aaf7e47f1a012d4564159193b767ff14eb0/extensions/ql-vscode/src/view/results/results.tsx#L93-L107) but they are called sequentially without any waiting, so there's no way to actually test that state.

Reply:
I'm gonna defer to @aeisenberg on this one. I would ask if we can get rid of `shouldKeepOldResultsWhileRendering` entirely if we're combining calls.

Reply:
Looks like Andrew is on holiday so you might have to wait a bit with this one.

Reply:
Just wanted to say that I wasn't expecting us to do a complete re-write of this view soon as the changes we'd like to make are mostly cosmetic. But I can be convinced otherwise.

Reply:
This method is overly complex and a simplification would be really nice. To be safe when we are doing a refactoring like this, it would be nice to add some pure regression unit tests to ensure that all the different cases are handled identically before and after.

Although nice, it's not necessary to spend the time to create more complex tests that check the view between the two `setState` calls. All that I think is necessary is that we ensure the method behaves the same before and after the refactoring. It's a complicated method, but at least the complexity is contained.

Reply:
I've created a really simple regression test in #2255. This does not test the loading state, but checks that we can at least render a SARIF file.